src/auth.rs

```diff
@@ -44,7 +47,7 @@ pub struct AuthQuery {
 impl FromStr for Service {
     type Err = ();
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
```

Comment:
Your crate's Result type might be generic, probably like 
```rust
type Result<T, E = crate::Error> = std::result::Result<T, E>;
```
which allows you to have two parameters but when you only have one it assumes it is your crate's error.